validate.py

```diff
@@ -53,8 +54,11 @@ def _pythons_to_check(tags: frozenset[Tag]) -> tuple[str, ...]:
             ret.add(_py_exe(*_parse_cp_tag(tag.interpreter)))
         elif tag.interpreter == "py2":
             continue
-        elif tag.interpreter == "py3":
-            ret.update(_py_exe(*py) for py in PYTHONS)
+        elif tag.interpreter.startswith("py3"):
+            for py in PYTHONS:
+                if tag not in packaging.tags.compatible_tags(py):
```

Comment:
@asottile-sentry is this an acceptable solution? for py 3.11 ci is building the wheel and ends up with py311-none-any tag - which afaict is within spec so we should do a broader compat check than == "py3"